enterprise/frontend/src/embedding-sdk/components/public/StaticQuestion/StaticQuestion.unit.spec.tsx

```diff
@@ -0,0 +1,162 @@
+import userEvent from "@testing-library/user-event";
+
+import {
+  setupCardEndpoints,
+  setupCardQueryEndpoints,
+  setupUnauthorizedCardEndpoints,
+} from "__support__/server-mocks";
+import {
+  renderWithProviders,
+  screen,
+  waitForLoaderToBeRemoved,
+  within,
+} from "__support__/ui";
+import { createMockConfig } from "embedding-sdk/test/mocks/config";
+import { createMockSdkState } from "embedding-sdk/test/mocks/state";
+import {
+  createMockCard,
+  createMockColumn,
+  createMockDataset,
+  createMockDatasetData,
+} from "metabase-types/api/mocks";
+import { createMockState } from "metabase-types/store/mocks";
+
+import type { QueryVisualizationProps } from "./";
+import { StaticQuestion } from "./";
+
+const TEST_QUESTION_ID = 1;
+const TEST_COLUMN = createMockColumn({
+  display_name: "Test Column",
+  name: "Test Column",
+});
+const TEST_DATASET = createMockDataset({
+  data: createMockDatasetData({
+    cols: [TEST_COLUMN],
+    rows: [["Test Row"]],
+  }),
+});
+
+const VISUALIZATION_TYPES: Record<
+  string,
+  {
+    container: string;
+    button: string;
+  }
+> = {
+  Table: { container: "Table-container", button: "Table-button" },
+  Number: {
+    container: "Number-container",
+    button: "Number-button",
+  },
+  Gauge: {
+    container: "Gauge-container",
+    button: "Gauge-button",
+  },
+  Detail: {
+    container: "Detail-container",
+    button: "Detail-button",
+  },
+  Progress: {
+    container: "Progress-container",
+    button: "Progress-button",
+  },
+};
+
+const setup = ({
+  showVisualizationSelector = false,
+  isValidCard = true,
+}: Partial<QueryVisualizationProps> & {
+  isValidCard?: boolean;
+} = {}) => {
+  const TEST_CARD = createMockCard();
+  if (isValidCard) {
+    setupCardEndpoints(TEST_CARD);
+  } else {
+    setupUnauthorizedCardEndpoints(TEST_CARD);
+  }
+  setupCardQueryEndpoints(TEST_CARD, TEST_DATASET);
+
+  // TODO: Do we need this here? We get a lot of warnings about how enterprise features aren't
+  // set up. We don't need them for this test though.
+  // Also, for whatever reason, we get a duplicate fake table when we don't use the SDK state - might just be a loading error or something since that fake table seems to be used to measure the cells for the layout (and should probably appear anyway). Might need to mock the TableInteractive component.
+  //
+  // const { state } = setupSdkState({
+  //   currentUser: TEST_USER,
+  // });
+
+  renderWithProviders(
+    <StaticQuestion
+      questionId={TEST_QUESTION_ID}
+      showVisualizationSelector={showVisualizationSelector}
+    />,
+    {
+      mode: "sdk",
+      sdkConfig: createMockConfig(),
+      storeInitialState: createMockState({
+        sdk: createMockSdkState({
+          loginStatus: { status: "success" },
+        }),
+      }),
+      // storeInitialState: state
```

Comment:
I think we should remove this commented line though

Reply:
Yeah I can live with the weirdness of the extra table for now. I'll remove these comments.